overlord/snapstate/storehelpers.go

```diff
@@ -781,6 +800,46 @@ func refreshCandidates(ctx context.Context, st *state.State, names []string, rev
 	return updates, stateByInstanceName, ignoreValidationByInstanceName, nil
 }
 
+// SnapHolds returns a map of held snaps to lists of holding snaps (including
+// "system" for user holds).
+func SnapHolds(st *state.State, snaps []string) (map[string][]string, error) {
+	var allSnapsHold string
+	tr := config.NewTransaction(st)
+	err := tr.Get("core", "refresh.hold", &allSnapsHold)
+	if err != nil && !config.IsNoOption(err) {
+		return nil, err
+	}
+
+	var allSnapsHoldTime time.Time
+	if allSnapsHold != "" {
+		if allSnapsHold == "forever" {
+			allSnapsHoldTime = timeNow().Add(maxDuration)
+		} else {
+			allSnapsHoldTime, err = time.Parse(time.RFC3339, allSnapsHold)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
```

Comment:
do we have code like this somewhere in the autorefresh code, that perhaps we should turn into a helper?

Reply:
I selected the wrong code range, see the remade comment below